impy/common.py

```diff
@@ -18,6 +18,28 @@
 from impy.util import info
 
 
+class EventData:
+    """Data structure to keep views of filtered data"""
+    def __init__(self, npart, p_ids, status, charge,
+                 px, py, pz, en, m, vx, vy, vz, vt,
+                 pem_arr, vt_arr):
+        self.npart = npart
```

Comment:
This looks like a derived quantity that should be a property?

```py
@property
def npart(self):
     return len(self.px)
```

Reply:
It can be a property if all other `px`, `py`, .. are properties as well. See below.

Reply:
Why? I don't see a reason to make everything a property here. You can use a property just for npart and leave the rest as is.

Reply:
Anyway, it is fine like this for this PR, I would like to experiment a bit with this class in a separate PR.